app-modules/campaign/src/Filament/Resources/CampaignResource/Pages/ListCampaigns.php

```diff
@@ -76,8 +78,17 @@ public function table(Table $table): Table
                     ->hidden(fn (Campaign $record) => $record->hasBeenExecuted() === true),
                 DeleteAction::make()
                     ->hidden(fn (Campaign $record) => $record->hasBeenExecuted() === true),
-            ])
-            ->bulkActions([
+            ])->filters([
+                Filter::make('My Campaigns')
+                    ->query(fn (Builder $query) => $query->where('user_id', auth()->user()->getKey())),
+                Filter::make('Enabled')
+                    ->query(fn (Builder $query) => $query->where('enabled', true)),
+                Filter::make('Completed')
```

Comment:
Let's adjust or create a `ListCampaignsTest` to test these filters and whether or not they work and display the records we would expect.

Reply:
@Orrison I've tried to create test case for filters, please let me know if I missed anything it's not correct.

Reply:
@kandarp-canyon unfortunately the tests and some of the logic was not correct. For sake of deployment though I have fixed them.